staging/src/k8s.io/client-go/tools/cache/thread_safe_store_test.go

```diff
@@ -165,6 +166,72 @@ func TestThreadSafeStoreIndexingFunctionsWithMultipleValues(t *testing.T) {
 	assert.NoError(compare("baz", []string{}))
 }
 
+func TestThreadSafeStoreByIndexes(t *testing.T) {
+	testIndexer1 := "test1Indexer"
+	testIndexer2 := "test2Indexer"
+
+	indexers := Indexers{
+		testIndexer1: func(obj interface{}) (strings []string, e error) {
+			indexes := []string{obj.(string)[:4]}
+			return indexes, nil
+		},
+		testIndexer2: func(obj interface{}) (strings []string, e error) {
+			indexes := []string{obj.(string)[4:5]}
+			return indexes, nil
+		},
+	}
+
+	indices := Indices{}
+	store := NewThreadSafeStore(indexers, indices).(*threadSafeMap)
+
+	store.Add(testIndexer1, testIndexer1)
+	store.Add(testIndexer2, testIndexer2)
+
+	tests := []struct {
+		name string
+		in   IndexConditions
+		out  []interface{}
+	}{
+		{
+			name: "equal match",
+			in: IndexConditions{
+				{Operator: selection.Equals, IndexName: testIndexer1, IndexKey: "test"},
+				{Operator: selection.Equals, IndexName: testIndexer2, IndexKey: "1"},
+			},
+			out: []interface{}{testIndexer1},
+		},
+		{
+			name: "double equal match",
+			in: IndexConditions{
+				{Operator: selection.DoubleEquals, IndexName: testIndexer1, IndexKey: "test"},
+				{Operator: selection.DoubleEquals, IndexName: testIndexer2, IndexKey: "1"},
+			},
+			out: []interface{}{testIndexer1},
+		},
+		{
+			name: "not equal match",
```

Comment:
It seems that this can be achieved with either of:
- `IndexKeys` + `ListKeys` + `Get`: get the obj keys for cond1 and cond2, get all the obj keys, figure out obj keys that are in cond1 and in (all - cond2), convert the keys to objects
- `ByIndex` + `List`: get the objects for cond1 and cond2, get all the objects, figure out objects that are in cond1 and in (all - cond2)

Could you explain why we need this `ByIndexes` method?

Reply:
Although several methods can be used to achieve the desired result, this method is sometimes too complicated for the user and prone to error.

Should we provide a simple interface for users to use?